packages/utils/src/internals/social.ts

```diff
@@ -681,13 +681,21 @@ export function parseHandlesFromHtml(html: string, data: Record<string, unknown>
     const text = htmlToText($);
     if (data) data.text = text;
 
+    // NOTE: we need to parse each text separately, orherwise we might concatenate unrelated texts
+    // e.g. `<div>6HT<a>eva@example.com</a></div>` would become 6HTeva@example.com
+    const texts = $('*')
+        .contents()
```

Comment:
Do I understand it right that this selects *all* elements and then it takes all the child nodes?

Reply:
yes, that's right, the point is to get all the text nodes in the HTML. I didn't find a better way to do it.

Reply:
it cannot, the results are later deduplicated [here](https://github.com/apify/crawlee/blob/279cadbd3cd6342f36cc4d841e07b999e472420d/packages/utils/src/internals/social.ts#L708)